ast/compile.go

```diff
@@ -1709,11 +1722,13 @@ func (ci *ComprehensionIndex) String() string {
 	return fmt.Sprintf("<keys: %v>", NewArray(ci.Keys...))
 }
 
-func buildComprehensionIndices(arity func(Ref) int, candidates VarSet, node interface{}, result map[*Term]*ComprehensionIndex) (n uint64) {
+func buildComprehensionIndices(dbg debug.Debug, arity func(Ref) int, candidates VarSet, node interface{}, result map[*Term]*ComprehensionIndex) uint64 {
+	var n uint64
 	WalkBodies(node, func(b Body) bool {
 		cpy := candidates.Copy()
 		for _, expr := range b {
-			if index := getComprehensionIndex(arity, cpy, expr); index != nil {
+			index := getComprehensionIndex(dbg, arity, cpy, expr)
+			if index != nil {
```

Comment:
I think it would be useful to include a message if a comprehension has was indexed (e.g., `<loc>: comprehension index built with keys: <keys>`). In practice, this doesn't affect most comprehensions so I think it will not be too noisy.

Reply:
✔️ I've put it in `getComprehensionIndex` right before the return.